cli/src/pcluster/imagebuilder_utils.py

```diff
@@ -65,3 +75,173 @@ def _generate_action(action_name, commands):
+                {"Action": "ec2:CreateTags", "Resource": f"arn:{partition}:ec2:*::image/*", "Effect": "Allow"},
+                {"Action": "tag:TagResources", "Resource": "*", "Effect": "Allow"},
+                {
+                    "Action": ["lambda:DeleteFunction", "lambda:RemovePermission"],
+                    "Resource": f"arn:{partition}:lambda:*:{account_id}:function:ParallelClusterImage-*",
+                    "Effect": "Allow",
+                },
+                {
+                    "Action": "logs:DeleteLogGroup",
+                    "Resource": f"arn:{partition}:logs:*:{account_id}:log-group:/aws/lambda/ParallelClusterImage-*:*",
+                    "Effect": "Allow",
+                },
+                {
+                    "Action": [
+                        "SNS:GetTopicAttributes",
+                        "SNS:DeleteTopic",
+                        "SNS:GetSubscriptionAttributes",
+                        "SNS:Unsubscribe",
+                    ],
+                    "Resource": f"arn:{partition}:sns:*:{account_id}:ParallelClusterImage-*",
+                    "Effect": "Allow",
+                },
+            ],
+        }
+    )
+
+
+def ensure_cleanup_role(account_id: str, partition: str = "aws", attach_vpc_access_policy: bool = False):
+    """
+    Ensure the global (account-wide) cleanup role exists and is at the expected revision.
+
+    The function follows a safe order:
+      1. If the role does not exist, create it without the revision tag.
+      2. If LambdaFunctionsVpcConfig exists in the config, attach the AWS-managed LambdaVPCAccess policy
+      3. Attach the AWS-managed Lambda basic policy.
+      4. Update/write the inline policy (least-privilege cleanup policy).
+      5. Only after the inline policy succeeds, set or bump the revision tag.
+
+    This way, if step 2 or 3 fails (e.g., lack of iam:PutRolePolicy permission) the tag
+    stays at an older revision and future invocations will keep retrying.
+    """
+    iam = AWSApi.instance().iam
+    role_name = get_cleanup_role_name(account_id)
+    role_arn = f"arn:{partition}:iam::{account_id}:role{IAM_ROLE_PATH}{role_name}"
+
+    # Assume-role trust policy
+    assume_doc = {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {"Service": "lambda.amazonaws.com"},
+                "Action": "sts:AssumeRole",
+                "Condition": {
+                    "ArnLike": {
+                        "aws:SourceArn": f"arn:{partition}:lambda:*:{account_id}:function:ParallelClusterImage-*"
+                    }
+                },
+            }
+        ],
+    }
+
+    # Check whether the role already exists
+    try:
+        resp = iam.get_role(role_name=role_name)
+        tags = {t["Key"]: t["Value"] for t in resp["Role"].get("Tags", [])}
+        current_revision = int(tags.get(CLEANUP_ROLE_REVISION_TAG_KEY, 0))
```

Comment:
why not using the revision number specified in PCLUSTER_BUILD_IMAGE_CLEANUP_ROLE_EXPECTED_REVISION?